nginx-tests-patches/2021-11-23-2786c65025940da977a65009e18aa3c5d23a92a6.patch

```diff
@@ -0,0 +1,5744 @@
+ stream_udp_limit_rate.t               |   1 +
+ stream_udp_proxy.t                    |   1 +
+ stream_udp_proxy_requests.t           |   1 +
+ stream_udp_stream.t                   |   1 +
+ stream_udp_upstream.t                 |   1 +
+ stream_udp_upstream_hash.t            |   1 +
+ stream_udp_upstream_least_conn.t      |   1 +
+ stream_udp_wildcard.t                 |   1 +
+ stream_unix.t                         |   1 +
+ stream_upstream.t                     |   1 +
+ stream_upstream_hash.t                |   1 +
+ stream_upstream_least_conn.t          |   1 +
+ stream_upstream_max_conns.t           |   1 +
+ stream_upstream_random.t              |   1 +
+ stream_upstream_zone.t                |   1 +
+ stream_upstream_zone_ssl.t            |   1 +
+ stream_variables.t                    |   1 +
+ stub_status.t                         |   1 +
+ sub_filter.t                          |   1 +
+ sub_filter_buffering.t                |   1 +
+ sub_filter_merge.t                    |   1 +
+ sub_filter_multi.t                    |   1 +
+ sub_filter_multi2.t                   |   1 +
+ sub_filter_perl.t                     |   1 +
+ sub_filter_slice.t                    |   1 +
+ sub_filter_ssi.t                      |   1 +
+ subrequest_output_buffer_size.t       |   1 +
+ syslog.t                              |   1 +
+ trailers.t                            |   1 +
+ upstream.t                            |   1 +
+ upstream_hash.t                       |   1 +
+ upstream_hash_memcached.t             |   1 +
+ upstream_ip_hash.t                    |   1 +
+ upstream_ip_hash_ipv6.t               |   1 +
+ upstream_keepalive.t                  |   1 +
+ upstream_least_conn.t                 |   1 +
+ upstream_max_conns.t                  |   1 +
+ upstream_random.t                     |   1 +
+ upstream_zone.t                       |   1 +
+ upstream_zone_ssl.t                   |   1 +
+ userid.t                              |   1 +
+ userid_flags.t                        |   1 +
+ uwsgi.t                               |   1 +
+ uwsgi_body.t                          |   1 +
+ uwsgi_ssl.t                           |   1 +
+ uwsgi_ssl_certificate.t               |   1 +
+ uwsgi_ssl_certificate_vars.t          |   1 +
+ uwsgi_ssl_verify.t                    |   1 +
+ xslt.t                                |   1 +
+ xslt_params.t                         |   1 +
+ 406 files changed, 598 insertions(+), 32 deletions(-)
+
+diff --git a/README b/README
+index f43c586..bd259ca 100644
+--- a/README
++++ b/README
+@@ -52,4 +52,12 @@ TEST_NGINX_GLOBALS_STREAM
+ 
+     Sets additional directives in stream context.
+ 
++TEST_NGINX_GDBSERVER
++
++    Run Nginx under a gdbserver.
++
++TEST_NGINX_VALGRIND
++
++    Run Nginx under valgrind.
++
+ Happy testing!
+diff --git a/access.t b/access.t
+index ab26ec0..af3ea5b 100644
+--- a/access.t
++++ b/access.t
+@@ -29,6 +29,7 @@ $t->write_file_expand('nginx.conf', <<'EOF');
+ 
+ %%TEST_GLOBALS%%
+ 
++master_process off;
```

Comment:
What's this do in the testing?

Reply:
This instructs Nginx to not fork itself into a master<->worker. With one process, it allows debugging through GDB. Multi-process debugging doesn't seam to work for me so this was necessary. Doesn't affect the tests except in a few cases (those do not have this instruction included).

Reply:
I would rather keep the patch to a minimum.
Please remove this in the files except for cases that we debug a lot.

Reply:
Removed `master_process off;`.